spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php

```diff
@@ -18,22 +18,51 @@ function it_is_initializable()
         $this->shouldHaveType(ErrorMaintainer::class);
     }
 
-    function it_return_false_when_error_suppresed_or_no_error_reporting()
+    function it_returns_false_when_error_suppresed_or_no_error_reporting()
     {
         $oldLevel = error_reporting(0);
+
         $this->errorHandler(0, 'error message', 'file', 1)->shouldBe(false);
+
+        error_reporting($oldLevel);
+    }
+
+    /**
+     * In PHP 8 error_reporting() will not return 0 anymore when
+     * and error is suppressed with the error control operator "@":
+     *
+     * @link https://www.php.net/manual/en/language.operators.errorcontrol.php
+     */
+    function it_returns_false_when_error_suppressed_in_php_8()
```

Comment:
No need to have `_in_php_8` in the name, because the same behavior should be present for other versions.

Reply:
I think it's useful to see the distinction because I'm making error reporting return this code, which is specific to php 8:

`error_reporting(E_ERROR | E_PARSE | E_CORE_ERROR | E_COMPILE_ERROR | E_USER_ERROR | E_RECOVERABLE_ERROR)`

The spec above it uses `error_reporting(0)`, which is the behaviour when suppressing errors in php < 8

Reply:
@kelunik Are you ok with this?